presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -243,16 +294,37 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             ColumnChunkMetaData metadata = getColumnChunkMetaData(columnDescriptor);
             long startingPosition = metadata.getStartingPos();
             int totalSize = toIntExact(metadata.getTotalSize());
-            byte[] buffer = allocateBlock(totalSize);
-            dataSource.readFully(startingPosition, buffer);
-            ColumnChunkDescriptor descriptor = new ColumnChunkDescriptor(columnDescriptor, metadata, totalSize);
-            ParquetColumnChunk columnChunk = new ParquetColumnChunk(descriptor, buffer, 0);
-            columnReader.init(columnChunk.readAllPages(), field);
 
-            if (enableVerification) {
-                ColumnReader verificationColumnReader = verificationColumnReaders[field.getId()];
-                ParquetColumnChunk columnChunkVerfication = new ParquetColumnChunk(descriptor, buffer, 0);
-                verificationColumnReader.init(columnChunkVerfication.readAllPages(), field);
+            if (shouldUseColumnIndex(metadata.getPath())) {
+                OffsetIndex offsetIndex = blockIndexStores.get(currentBlock).getOffsetIndex(metadata.getPath());
+                OffsetIndex filteredOffsetIndex = ColumnIndexFilterUtils.filterOffsetIndex(offsetIndex, currentGroupRowRanges, blocks.get(currentBlock).getRowCount());
+                List<OffsetRange> offsetRanges = ColumnIndexFilterUtils.calculateOffsetRanges(filteredOffsetIndex, metadata, offsetIndex.getOffset(0), startingPosition);
+                List<ConsecutivePartList> allParts = concatRanges(offsetRanges);
```

Comment:
s/allParts/offsetRanges/g

Reply:
We already have the variable offsetRanges. We concat it.

Reply:
We already have the variable offsetRanges. We concat it.

Reply:
get it. how about:
`consecutiveRanges`?
generally, we do not use variable abbreviation in Presto

Reply:
Changed it.